test/test_euler.py

```diff
@@ -660,6 +708,76 @@ def test_lump_rhs(actx_factory, dim, order):
     )
 
 
+@pytest.mark.parametrize("dim", [1, 2, 3])
+@pytest.mark.parametrize("order", [1, 2, 4])
+@pytest.mark.parametrize("nspecies", [1, 2, 10])
+@pytest.mark.parametrize("v0", [0.0, 1.0])
+def test_multilump_rhs(actx_factory, dim, order, nspecies, v0):
+    """Tests the inviscid rhs using the non-trivial
+    1, 2, and 3D mass lump case against the analytic
+    expressions of the RHS. Checks several different
+    orders and refinement levels to check error behavior.
+    """
+    actx = actx_factory()
+
+    tolerance = 1e-8
+    maxxerr = 0.0
+
+    from pytools.convergence import EOCRecorder
+
+    eoc_rec = EOCRecorder()
+
+    for nel_1d in [4, 8, 16]:
+        from meshmode.mesh.generation import (
+            generate_regular_rect_mesh,
+        )
+
+        mesh = generate_regular_rect_mesh(
+            a=(-1,) * dim, b=(1,) * dim, n=(nel_1d,) * dim,
+        )
+
+        logger.info(f"Number of elements: {mesh.nelements}")
+
+        discr = EagerDGDiscretization(actx, mesh, order=order)
+        nodes = thaw(actx, discr.nodes())
+
+        centers = make_obj_array([np.zeros(shape=(dim,)) for i in range(nspecies)])
+        spec_y0s = np.ones(shape=(nspecies,))
+        spec_amplitudes = np.ones(shape=(nspecies,))
+
+        velocity = np.zeros(shape=(dim,))
+        velocity[0] = v0
+
+        lump = MultiLump(numdim=dim, nspecies=nspecies, spec_centers=centers,
+                         velocity=velocity, spec_y0s=spec_y0s,
+                         spec_amplitudes=spec_amplitudes)
+
+        lump_soln = lump(t=0, x_vec=nodes)
+        boundaries = {BTAG_ALL: PrescribedBoundary(lump)}
+
+        inviscid_rhs = inviscid_operator(
+            discr, eos=IdealSingleGas(), boundaries=boundaries, q=lump_soln, t=0.0)
+        expected_rhs = lump.exact_rhs(discr, lump_soln, 0)
+        print(f"inviscid_rhs = {inviscid_rhs}")
+        print(f"expected_rhs = {expected_rhs}")
+        err_max = discr.norm(inviscid_rhs-expected_rhs, np.inf)
+        if err_max > maxxerr:
+            maxxerr = err_max
+
+        eoc_rec.add_data_point(1.0 / nel_1d, err_max)
+    logger.info(f"Max error: {maxxerr}")
+
+    message = (
+        f"Error for (dim,order) = ({dim},{order}):\n"
+        f"{eoc_rec}"
+    )
+    logger.info(message)
```

Comment:
Why a separate variable?

Reply:
Because I (and others) cannot figure out how to see our output from tests when it goes into this `logger`.  If I stick it in a message string, then I can also `print` the message.  I don't want to set the logger level so that I see miles of extraneous output before I get to the thing I really want to see.